packages/fx-core/src/component/generator/generatorAction.ts

```diff
@@ -117,26 +120,25 @@ export const downloadDirectoryAction: GeneratorAction = {
 };
 
 export const fetchUrlForHotfixOnlyAction: GeneratorAction = {
-  name: GeneratorActionName.FetchTemplateUrlWithTag,
+  name: GeneratorActionName.FetchUrlForHotfixOnly,
   run: async (context: GeneratorContext) => {
-    if (context.zip || context.url || context.cancelDownloading) {
+    if (context.zip || context.cancelDownloading) {
       return;
     }
 
     context.logProvider.debug(`Fetching template url with tag: ${JSON.stringify(context)}`);
-    const url = await fetchTemplateZipUrl(
+    const latestTag = await getTemplateLatestTag(
       context.language!,
       context.tryLimits,
       context.timeoutInMs
     );
-    let parts = url.split("/");
-    const selectedTag = parts[parts.length - 2];
-    parts = selectedTag.split(".");
-    // hotfix generates template based on url
-    if (parts[parts.length - 1] !== "0") {
-      context.url = url;
+    const localVer = getTemplateLocalVersion();
+    const latestVer = latestTag.split("@")[1];
+    // git tag version is higher than the local version, download template from github
+    if (semver.gt(latestVer, localVer)) {
+      context.url = getTemplateZipUrlByTag(context.language!, latestTag);
     } else {
-      // stable generates templates based on the fallback
+      // download template from fallback
```

Comment:
The comment is inaccurate. We don't `download` template when using fallback.

Reply:
I'll use "generate" instead of "download".